crates/bevy_asset/src/asset_server.rs

```diff
@@ -500,11 +506,16 @@ impl AssetServer {
     // triggered unless the `Assets` collection is actually updated.
     pub(crate) fn update_asset_storage<T: Asset>(&self, mut assets: ResMut<Assets<T>>) {
         let asset_lifecycles = self.server.asset_lifecycles.read();
-        let asset_lifecycle = asset_lifecycles.get(&T::TYPE_UUID).unwrap();
+        let asset_lifecycle = asset_lifecycles.get(&T::TYPE_UUID).expect_or_else(|| {
+            format!(
+                "Could not find an asset of the appropriate type UUID={}.",
+                T::TYPE_UUID
+            )
+        });
```

Comment:
```suggestion
        let asset_lifecycle = asset_lifecycles.get(&T::TYPE_UUID).unwrap_or_else(|| {
            panic!(
                "Could not find an asset of the appropriate type UUID={}.",
                T::TYPE_UUID
            )
        });
```

Here's the actual line that compiles